compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -627,6 +618,29 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
       }
     }
 
+    private def reportCannotDisambiguateError(alts: List[Exported]): Unit = {
+      val currentClass = currentClassSym.get
+
+      // Find a position that is in the current class for decent error reporting
+      val pos = alts.collectFirst {
+        case ExportedSymbol(sym) if sym.owner == currentClass => sym.pos
+        case alt: ExportedBody                                => alt.pos
+      }.getOrElse {
+        currentClass.pos
+      }
```

Comment:
> Hmmm... Probably not, since it doesn't really generate any real methods. So a trait never actually defines any method.

Indeed. The above code is valid.

> Alternatively we could put the assert and still fallback, so a user that gets hit by this could disable the asserts (although we don't have a precedent for this actually happening).

It's not possible to disable a Scala `assert`. Not after translation to bytecode, anyway. This is different from Java asserts, which have run-time support in the VM.